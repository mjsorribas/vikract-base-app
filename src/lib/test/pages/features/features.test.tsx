import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Features from "lib/pages/features/index";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Features Page", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/features" }]}>
      <Features />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
