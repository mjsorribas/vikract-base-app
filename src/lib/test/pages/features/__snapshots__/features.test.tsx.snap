// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Features Page 1`] = `
<div
  className="css-qi60vh"
>
  <h2
    className="chakra-heading css-0"
  >
    Dynamic Modal
  </h2>
  <p
    className="chakra-text css-0"
  >
    Vikract core includes custom components to simplified the developer life.
    <br />
    <ul
      className="css-me00nr"
      role="list"
    >
      <li
        className="css-0"
      >
        Carousel
      </li>
      <li
        className="css-0"
      >
        Contact Form
      </li>
      <li
        className="css-0"
      >
        Price Card
      </li>
      <li
        className="css-0"
      >
        Dynamic Tabs
      </li>
      <li
        className="css-0"
      >
        <a
          className="chakra-link css-1856l8r"
          href="/features/dynamicmodal"
        >
          Dynamic Modal
        </a>
      </li>
      <li
        className="css-0"
      >
        ActionButton
      </li>
      <li
        className="css-0"
      >
        Card
      </li>
    </ul>
  </p>
</div>
`;
